src/envs/coffee.py

```diff
@@ -0,0 +1,756 @@
+    init_padding: ClassVar[float] = 0.5  # used to space objects in init states
+    pick_jug_y_padding: ClassVar[float] = 0.5
+    # Robot settings.
+    x_lb: ClassVar[float] = 0.0
+    x_ub: ClassVar[float] = 10.0
+    y_lb: ClassVar[float] = 0.0
+    y_ub: ClassVar[float] = 10.0
+    z_lb: ClassVar[float] = 0.0
+    z_ub: ClassVar[float] = 10.0
+    tilt_lb: ClassVar[float] = 0.0
+    tilt_ub: ClassVar[float] = np.pi / 4
+    robot_init_x: ClassVar[float] = (x_ub + x_lb) / 2.0
+    robot_init_y: ClassVar[float] = (y_ub + y_lb) / 2.0
+    robot_init_z: ClassVar[float] = z_ub
+    robot_init_tilt: ClassVar[float] = 0.0
+    open_fingers: ClassVar[float] = 0.4
+    closed_fingers: ClassVar[float] = 0.1
+    # Machine settings.
+    machine_x_len: ClassVar[float] = 0.1 * (x_ub - x_lb)
+    machine_y_len: ClassVar[float] = 0.2 * (y_ub - y_lb)
+    machine_z_len: ClassVar[float] = 0.4 * (z_ub - z_lb)
+    machine_x: ClassVar[float] = x_ub - machine_x_len - init_padding
+    machine_y: ClassVar[float] = y_ub - machine_y_len - init_padding
+    button_x: ClassVar[float] = machine_x + machine_x_len / 2
+    button_y: ClassVar[float] = machine_y
+    button_z: ClassVar[float] = 3 * machine_z_len / 4
+    button_radius: ClassVar[float] = 0.2 * machine_x_len
+    # Jug settings.
+    jug_radius: ClassVar[float] = (0.8 * machine_x_len) / 2.0
+    jug_height: ClassVar[float] = 0.15 * (z_ub - z_lb)
+    jug_init_x_lb: ClassVar[float] = machine_x - machine_x_len + init_padding
+    jug_init_x_ub: ClassVar[float] = machine_x + machine_x_len - init_padding
+    jug_init_y_lb: ClassVar[float] = y_lb + jug_radius + init_padding
+    jug_init_y_ub: ClassVar[
+        float] = machine_y - machine_y_len - jug_radius - init_padding
+    jug_handle_x_offset: ClassVar[float] = 0.0
+    jug_handle_y_offset: ClassVar[float] = -(1.05 * jug_radius)
+    jug_handle_height: ClassVar[float] = 3 * jug_height / 4
+    # Dispense area settings.
+    dispense_area_x: ClassVar[float] = machine_x + machine_x_len / 2
+    dispense_area_y: ClassVar[float] = machine_y - 1.1 * jug_radius
+    # Cup settings.
+    cup_radius: ClassVar[float] = 0.6 * jug_radius
+    cup_init_x_lb: ClassVar[float] = x_lb + cup_radius + init_padding
+    cup_init_x_ub: ClassVar[
+        float] = machine_x - machine_x_len - cup_radius - init_padding
+    cup_init_y_lb: ClassVar[float] = jug_init_y_lb
+    cup_init_y_ub: ClassVar[float] = jug_init_y_ub
+    cup_capacity_lb: ClassVar[float] = 0.075 * (z_ub - z_lb)
+    cup_capacity_ub: ClassVar[float] = 0.15 * (z_ub - z_lb)
+    cup_target_frac: ClassVar[float] = 0.75  # fraction of the capacity
+    # Simulation settings.
+    pour_x_offset: ClassVar[float] = 1.5 * (cup_radius + jug_radius)
+    pour_y_offset: ClassVar[float] = cup_radius
+    pour_z_offset: ClassVar[float] = 1.1 * (cup_capacity_ub + jug_height - \
+                                            jug_handle_height)
+    pour_velocity: ClassVar[float] = cup_capacity_ub / 10.0
```

Comment:
so this `/ 10.0` means it'll take on the order of 10 timesteps to finish pouring? (i know it's actually less because target is 3/4 the capacity)

Reply:
yeah